src/services/package-manager/batch-import-export-service.ts

```diff
@@ -0,0 +1,38 @@
+import {batchImportExportApi} from "../../api/batch-import-export-api";
+import {logger} from "../../util/logger";
+import {v4 as uuidv4} from "uuid";
+import {PackageExportTransport} from "../../interfaces/package-export-transport";
+import {FileService, fileService} from "../file-service";
+import {studioService} from "../studio/studio.service";
+
+class BatchImportExportService {
+
+    public async listActivePackages(flavors: string[]): Promise<void> {
+        const activePackages = await batchImportExportApi.findAllActivePackages(flavors);
+        activePackages.forEach(pkg => {
+            logger.info(`${pkg.name} - Key: "${pkg.key}"`)
+        });
+    }
+
+    public async findAndExportListOfActivePackages(flavors: string[], packageKeys: string[], withDependencies: boolean): Promise<void> {
+        let packagesToExport: PackageExportTransport[];
+
+        if (packageKeys.length) {
+            packagesToExport = await batchImportExportApi.findActivePackagesByKeys(packageKeys, withDependencies);
+        } else  {
+            packagesToExport = await batchImportExportApi.findAllActivePackages(flavors, withDependencies);
+        }
+
+        packagesToExport = await studioService.getExportPackagesWithStudioData(packagesToExport, withDependencies);
+
+        this.exportListOfPackages(packagesToExport);
+    }
+
+    private exportListOfPackages(packages: PackageExportTransport[]): void {
+        const filename = uuidv4() + ".json";
```

Comment:
could we use a more descriptive name for the json file instead of a UUID?

Reply:
In that case at the very list adding a specific static prefix would make the generated files a lot easier to read, for example packages-<uuid>

Reply:
All other listing commands use randomized naming pattern. I would leave this as is and tackle them all at once if we decide to change this.